plugins/plugin_utils/connectivity_utils.py

```diff
@@ -0,0 +1,373 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import time
+import os
+import json
+import base64
+import tarfile
+from glob import glob
+import traceback
+
+HAS_ORAS = False
+HAS_ORAS_EXC = None
+
+try:
+    import oras.client
+    HAS_ORAS = True
+except ImportError:
+    oras = None
+    HAS_ORAS_EXC = traceback.format_exc()
+
+try:
+    from azure.core.exceptions import ResourceNotFoundError, HttpResponseError
+    from azure.mgmt.core.tools import resource_id, parse_resource_id
+except ImportError:
+    pass
+
+from ansible_collections.azure.azcollection.plugins.plugin_utils import file_utils
+from ansible_collections.azure.azcollection.plugins.plugin_utils import constants as consts
+
+from ansible.errors import AnsibleParserError, AnsibleError
```

Comment:
Unused AnsibleParserError imported from ansible.errors
```suggestion
from ansible.errors import AnsibleError
```